internal/webhooks/cluster_test.go

```diff
@@ -51,6 +52,144 @@ func TestClusterDefaultNamespaces(t *testing.T) {
+				Build(),
+			clustervars: []clusterv1.ClusterVariable{},
+			expect: []clusterv1.ClusterVariable{
+				{
+					Name: "location",
+					Value: apiextensionsv1.JSON{
+						Raw: []byte(`"us-east"`),
+					},
+				},
+			},
+		},
+		{
+			name: "don't change a variable if it is already set",
+			clusterclass: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+				WithVariables([]clusterv1.ClusterClassVariable{
+					{
+						Name:     "location",
+						Required: true,
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type:    "string",
+								Default: &apiextensionsv1.JSON{Raw: []byte(`"us-east"`)},
+							},
+						},
+					},
+				}).
+				Build(),
+			clustervars: []clusterv1.ClusterVariable{
+				{
+					Name: "location",
+					Value: apiextensionsv1.JSON{
+						Raw: []byte(`"A different location"`),
+					},
+				},
+			},
+			expect: []clusterv1.ClusterVariable{
+				{
+					Name: "location",
+					Value: apiextensionsv1.JSON{
+						Raw: []byte(`"A different location"`),
+					},
+				},
+			},
+		},
+		{
+			name: "default many variables to its correct values",
+			clusterclass: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+				WithVariables([]clusterv1.ClusterClassVariable{
+					{
+						Name:     "location",
+						Required: true,
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type:    "string",
+								Default: &apiextensionsv1.JSON{Raw: []byte(`"us-east"`)},
+							},
+						},
+					},
+					{
+						Name:     "count",
+						Required: true,
+						Schema: clusterv1.VariableSchema{
+							OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+								Type:    "number",
+								Default: &apiextensionsv1.JSON{Raw: []byte(`0.1`)},
+							},
+						},
+					},
+				}).
+				Build(),
+			clustervars: []clusterv1.ClusterVariable{},
+			expect: []clusterv1.ClusterVariable{
+				{
+					Name: "location",
+					Value: apiextensionsv1.JSON{
+						Raw: []byte(`"us-east"`),
+					},
+				},
+				{
+					Name: "count",
+					Value: apiextensionsv1.JSON{
+						Raw: []byte(`0.1`),
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		cluster := builder.Cluster(metav1.NamespaceDefault, "cluster1").
+			WithTopology(
+				builder.ClusterTopology().
+					WithClass("class1").
+					WithVersion("v1.22.2").
+					WithVariables(tt.clustervars).
+					Build()).
+			Build()
+		fakeClient := fake.NewClientBuilder().
+			WithObjects(tt.clusterclass).
+			WithScheme(fakeScheme).
+			Build()
+
+		// Create the webhook and add the fakeClient as its client. This is required because the test uses a Managed Topology.
+		webhook := &Cluster{Client: fakeClient}
+		t.Run(tt.name, customDefaultValidateTest(ctx, cluster, webhook))
+		t.Run(tt.name, func(t *testing.T) {
+			err := webhook.Default(ctx, cluster)
+			g.Expect(err).To(Not(HaveOccurred()))
+			g.Expect(cluster.Spec.Topology.Variables).To(Equal(tt.expect))
+		})
```

Comment:
```suggestion
		t.Run(tt.name, func(t *testing.T) {
			// Test if defaulting works in combination with validation.
			customDefaultValidateTest(ctx, cluster, webhook)(t)

			// Test defaulting.
			t.Run("default", func(t *testing.T) {
				g := NewWithT(t)
				g.Expect(webhook.Default(ctx, cluster)).To(Succeed())
				g.Expect(cluster.Spec.Topology.Variables).To(Equal(tt.expect))
			})
		})
```
WDYT about using this instead. This changes the test hierarchy from:

![image](https://user-images.githubusercontent.com/4662360/143266712-184c9932-57d7-42fa-8eb7-a81a56138ef9.png)

to:
![image](https://user-images.githubusercontent.com/4662360/143266752-c29cd5e7-364e-4d0c-9329-eb898f0b1f49.png)

Note: `g := NewWithT(t)` is important in any case. (the one in l.58 should be dropped)